README.md

```diff
@@ -0,0 +1,50 @@
+
+# SI-2712 fix plugin
+
+This is proof of concept fix to [SI-2712][si2712] implemented as a compiler plugin. There is also a [PR][si2712pr] to fix directly in scalac.
+The implementation is based on a simple algorithm as suggested by Paul Chiusano in the comments on [SI-2712][si2712]. Treat the type constructor as curried and partially applied, we treat a prefix as constants and solve for the suffix. For the example in the ticket, unifying M[A] with Int => Int this unifies as,
+
+```Scala
+M[t] = [t][Int => t]
+A = Int
+```
+More detailed explanations can also be found at this gist [Explaining Miles's Magic][explain] by @djspiewak and the readme of the [demo project][demo].
+
+
+
+## Usage
+
+For now you have to build the plugin yourself. Clone the project and run
+```bash
+sbt publishLocal
```

Comment:
Maybe `+ publishLocal` because as of #2 it's on both Scala 2.10 and Scala 2.11 ;)